imap_processing/ultra/l2/ultra_l2.py

```diff
@@ -0,0 +1,483 @@
+            flat_indices_dps
+        ]
+        if is_ultra45(l1c_prod):
+            combined_exptime_45 += pointing_projected_exptime
+        else:
+            combined_exptime_90 += pointing_projected_exptime
+
+    combined_exptime_total = combined_exptime_45 + combined_exptime_90
+
+    return (combined_exptime_45, combined_exptime_90, combined_exptime_total)
+
+
+@typing.no_type_check
+@ensure_spice
+def build_flux_maps(
+    l1c_products: list[xr.Dataset], spacing_deg: float = DEFAULT_SPACING_DEG
+) -> tuple[NDArray]:
+    """
+    Build maps of particle counts, flux, etc. from multiple l1c observations.
+
+    Parameters
+    ----------
+    l1c_products : list[xr.Dataset]
+        List of l1c product datasets.
+    spacing_deg : float, optional
+        The spacing in degrees of the output maps, by default DEFAULT_SPACING_DEG.
+
+    Returns
+    -------
+    tuple[NDArray]
+        Tuple of the following arrays:
+        - Combined counts for each energy bin
+        - Combined exposure time for both sensors
+        - Combined exposure time for 45 degree sensor
+        - Combined exposure time for 90 degree sensor
+        - Frame epochs
+        - Energy bin widths
+        - Energy bin edges
+        - Energy bin midpoints
+        - Solid angle grid
+        - Developer dictionary containing intermediate results
+        NOTE: Many of these are temporary, used in dev, and will be removed.
+
+    Raises
+    ------
+    ValueError
+        If the az, el, or energy bin centers don't match across l1c products.
+    ValueError
+        _description_
+    """
+    # Check that all l1c products have the same shape of the az, el, energy bin coords
+    for l1c_product in l1c_products:
+        if (
+            not l1c_product.azimuth_bin_center.values.shape
+            == l1c_products[0].azimuth_bin_center.values.shape
+        ):
+            raise ValueError("Azimuth bin center shape mismatch.")
+        if (
+            not l1c_product.elevation_bin_center.values.shape
+            == l1c_products[0].elevation_bin_center.values.shape
+        ):
+            raise ValueError("Elevation bin center shape mismatch.")
+        if (
+            not l1c_product.energy_bin_center.values.shape
+            == l1c_products[0].energy_bin_center.values.shape
+        ):
+            raise ValueError("Energy bin center shape mismatch.")
+
+    (num_el, num_az, num_energy) = l1c_products[0].counts.shape
+
+    # This will contain the counts for each energy bin
+    combined_counts = np.zeros((num_el * num_az, num_energy))
+
+    # Build the azimuth and elevation grid in the heliocentric ecliptic frame (HAE)
+    (az_range, el_range, az_grid, el_grid) = l2_utils.build_az_el_grid(
+        spacing=spacing_deg,
+        input_degrees=True,
+        output_degrees=False,
+        centered_azimuth=False,  # (0, 2pi rad = 0, 360 deg)
+        centered_elevation=True,  # (-pi/2, pi/2 rad = -90, 90 deg)
```

Comment:
I see that these grids are build both in this function as well as in the `build_dps_combined_exposure_time` function. Can this duplication be removed somehow?